sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -76,7 +76,7 @@
    * checkpoint if present.
    */
   public abstract UnboundedReader<OutputT> createReader(
-      PipelineOptions options, @Nullable CheckpointMarkT checkpointMark);
+      PipelineOptions options, @Nullable CheckpointMarkT checkpointMark) throws IOException;
```

Comment:
Can you avoid this API change? Should you?

@dpmills 

[if you don't, we'll have to diverge if we decide to port this back to Dataflow]